cpp/src/arrow/flight/client.h

```diff
@@ -129,6 +133,12 @@ class ARROW_FLIGHT_EXPORT FlightStreamReader : public MetadataRecordBatchReader
  public:
   /// \brief Try to cancel the call.
   virtual void Cancel() = 0;
+  using MetadataRecordBatchReader::ReadAll;
+  /// \brief Consume entire stream as a vector of record batches
+  virtual Status ReadAll(std::vector<std::shared_ptr<RecordBatch>>* batches,
+                         const StopToken& stop_token) = 0;
+  /// \brief Consume entire stream as a Table
+  virtual Status ReadAll(std::shared_ptr<Table>* table, const StopToken& stop_token) = 0;
```

Comment:
Is there a reason for the `Table` version to be virtual? It seems it can simply be implemented in the base class...?

Of course, it doesn't matter much as long as we have only one implementation of this.

Reply:
I made this non-virtual.